### PR TITLE
Removes sleeps and use Eventually construct in integration tests

### PR DESCRIPTION
**Pull request by suprjinx**

### CODE REVIEW

**Comment by kannon92** [comment]
https://github.com/armadaproject/armada-operator/issues/74

File: test/integration/armadaserver_controller_test.go [hunk 1]
```diff
@@ -36,29 +35,31 @@ var _ = Describe("Armada Operator", func() {
...
 
-				time.Sleep(2 * time.Second)
-
 				as := installv1alpha1.ArmadaServer{}
 				asKey := kclient.ObjectKey{Namespace: "default", Name: "armadaserver-e2e"}
-				err = k8sClient.Get(ctx, asKey, &as)
-				Expect(err).NotTo(HaveOccurred())
+				Eventually(func() error {
+					return k8sClient.Get(ctx, asKey, &as)
+				}, "2s", "10ms").ShouldNot(HaveOccurred())
```
Comment (kannon92) [inline test/integration/armadaserver_controller_test.go:42]: Could we refactor that as a constant so it would be easy to update these in one place?

**Comment by ClifHouck** [comment]
My only comment is that are we sure 2s is enough of a timeout for all these operations?

File: test/integration/armadaserver_controller_test.go (hunk 1 above)
Comment (suprjinx) [reply test/integration/armadaserver_controller_test.go:42]: okay will do

**Review by ClifHouck — APPROVED** [review]

**Pull request merged**